### PR TITLE
[AAE-10309] - Joy to the world the unit tests are fixed!

### PR DESCRIPTION
**Pull request by VitoAlbano**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Review by eromano — APPROVED** [review]

File: lib/process-services-cloud/src/lib/form/components/form-cloud.component.spec.ts [hunk 1]
```diff
@@ -974,6 +972,10 @@ describe('FormCloudComponent', () => {
 
         formComponent.form = new FormModel();
```
Comment (mauriziovitale) [inline lib/process-services-cloud/src/lib/form/components/form-cloud.component.spec.ts:973]: the form should be instantiated through the detect change not manually, same for the outcome

File: lib/process-services-cloud/src/lib/process/start-process/components/start-process-cloud.component.spec.ts [hunk 2]
```diff
@@ -206,17 +206,19 @@ describe('StartProcessCloudComponent', () => {
...
 
     describe('start a process with start form', () => {
 
         beforeEach(() => {
             component.name = 'My new process with form';
+            component.appName = 'startformwithoutupload';
+            getDefinitionsSpy.and.returnValue(of(fakeProcessDefinitions));
+            fixture.detectChanges();
+            const change = new SimpleChange(null, 'startformwithoutupload', true);
+            component.ngOnChanges({ appName: change });
```
[the remark this answers is not in the document]
Comment (VitoAlbano) [reply lib/process-services-cloud/src/lib/process/start-process/components/start-process-cloud.component.spec.ts:221]: Nope ngOnChanges is never triggered on the fixture.detectChanges() :D unfortunately angular decision

File: lib/process-services-cloud/src/lib/form/components/form-cloud.component.spec.ts (hunk 1 above)
Comment (VitoAlbano) [reply lib/process-services-cloud/src/lib/form/components/form-cloud.component.spec.ts:973]: In this case does it really make sense? we are checking that the event is emitted.

File: lib/process-services-cloud/src/lib/process/start-process/components/start-process-cloud.component.spec.ts [hunk 3]
```diff
@@ -239,218 +241,174 @@ describe('StartProcessCloudComponent', () => {
...
 
-        it('should be able to start a process with a prefilled valid form', fakeAsync(() => {
-            component.processDefinitionName = 'processwithform';
-            getDefinitionsSpy.and.returnValue(of(fakeSingleProcessDefinition(component.processDefinitionName)));
-            fixture.detectChanges();
-            formDefinitionSpy = spyOn(formCloudService, 'getForm').and.returnValue(of(fakeStartForm));
-
-            const change = new SimpleChange(null, 'MyApp', true);
-            component.ngOnChanges({ appName: change });
             fixture.detectChanges();
```
[the remark this answers is not in the document]
Comment (VitoAlbano) [reply lib/process-services-cloud/src/lib/process/start-process/components/start-process-cloud.component.spec.ts:273]: If i remember correctly those are there to go out of the loading state

File: lib/process-services-cloud/src/lib/process/start-process/components/start-process-cloud.component.spec.ts [hunk 4]
```diff
@@ -239,218 +241,174 @@ describe('StartProcessCloudComponent', () => {
...
 
-        it('should NOT be able to start a process with a prefilled NOT valid form', fakeAsync(() => {
-            component.processDefinitionName = 'processwithform';
-            getDefinitionsSpy.and.returnValue(of(fakeSingleProcessDefinition(component.processDefinitionName)));
-            fixture.detectChanges();
-            formDefinitionSpy = spyOn(formCloudService, 'getForm').and.returnValue(of(fakeStartFormNotValid));
-
-            const change = new SimpleChange(null, 'MyApp', true);
-            component.ngOnChanges({ appName: change });
             fixture.detectChanges();
```
[the remark this answers is not in the document]
Comment (VitoAlbano) [reply lib/process-services-cloud/src/lib/process/start-process/components/start-process-cloud.component.spec.ts:291]: as above.

**Review by DenysVuika — APPROVED** [review]

**Pull request merged**